src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -214,13 +223,13 @@ template <size_t MaxDeriv>
 std::ostream& operator<<(
     std::ostream& os,
     const PiecewisePolynomial<MaxDeriv>& piecewise_polynomial) {
```

Comment:
I'm not sure this is threadsafe. Haven't thought about it much, but I worry that calling `end` or `size` could be bad. I'm also not sure we care

Reply:
Definitely not thread-safe. But I also don't think we care. I can easily change it to be thread-safe though